lambdas/check-existing-identity/src/main/java/uk/gov/di/ipv/core/checkexistingidentity/CheckExistingIdentityHandler.java

```diff
@@ -410,4 +417,102 @@ private List<String> extractTxnIdsFromCredentials(List<SignedJWT> credentials)
         }
         return txnIds;
     }
+
+    @Tracing
+    private Optional<VotProfilePair> getAttainedVotAndProfile(
+            List<String> vtr,
+            List<SignedJWT> credentials,
+            List<VcStoreItem> vcStoreItems,
+            AuditEventUser auditEventUser)
+            throws UnknownEvidenceTypeException, ParseException, SqsException {
+
+        var requestedVotsByStrength =
+                SUPPORTED_VOTS_BY_STRENGTH.stream()
+                        .filter(vot -> vtr.contains(vot.name()))
+                        .toList();
+
+        for (var requestedVot : requestedVotsByStrength) {
+            var attainedVotAndProfile =
+                    requestedVot.hasGpg45Profiles()
+                            ? matchGpg45Profile(
+                                    credentials, vcStoreItems, requestedVot, auditEventUser)
+                            : matchOperationalProfile(credentials, requestedVot);
+
+            if (attainedVotAndProfile.isPresent()) {
+                LOGGER.info(
+                        new StringMapMessage()
+                                .with(
+                                        LOG_MESSAGE_DESCRIPTION.getFieldName(),
+                                        "Attained vot with profile")
+                                .with(
+                                        LOG_VOT.getFieldName(),
+                                        attainedVotAndProfile.get().attainedVot())
+                                .with(
+                                        LOG_PROFILE.getFieldName(),
+                                        attainedVotAndProfile.get().profileName()));
+
+                return attainedVotAndProfile;
+            }
+        }
+        return Optional.empty();
+    }
+
+    private Optional<VotProfilePair> matchGpg45Profile(
+            List<SignedJWT> credentials,
+            List<VcStoreItem> vcStoreItems,
```

Comment:
It's pretty suspicious that this requires both `credentials` and `vcStoreItems`, but not sure if you can easily untangle this.

Feels like it _should_ just be `credentials`, but `checkRequiresAdditionalEvidence` uses the CRI Id instead :/

Something I'd like to fix as part of the proposed VC model refactoring, so worry _too_ much about it now.

Reply:
Yeah, this annoyed me. I didn't want to have to parse all the credentials again just to reduce the number of params.
I think it would be interesting to understand exactly how heavy a process that is to do...

Reply:
I don't think it's very heavy - string splitting and presumably base64 decoding. Ideally I think we'd have one model class, which handles the 'parse once' logic, but not worth doing now.

In the meantime, I don't mind too much - either re-parse or have two args 🤷